p2p/p2p.go

```diff
@@ -244,7 +245,7 @@ func (s *Service) Run(ctx context.Context) error {
 		return err
 	}
 
-	s.pubsub, err = pubsub.NewGossipSub(ctx, s.host)
+	s.pubsub, err = pubsub.NewGossipSub(ctx, s.host, pubsub.WithRawTracer(s.gossipTracer))
```

Comment:
if caller doesn't call `WithGossipTracer()` property `gossipTracer` is nil which will lead to:
```
return func(p *PubSub) error {
		if p.tracer != nil {
			p.tracer.raw = append(p.tracer.raw, tracer)
		} else {
                        // this branch
			p.tracer = &pubsubTracer{raw: []RawTracer{tracer}, pid: p.host.ID(), idGen: p.idGen}
		}
		return nil
	}
```
so I think we should pass this option only if `s.gossipTracer` is not nil, something like this:
```
var options []GossipOption
if s.gossipTracer != nil {
    options = append(options, pubsub.WithRawTracer(s.gossipTracer))
}
s.pubsub, err = pubsub.NewGossipSub(ctx, s.host, options...)
```